server/backend/api/app/crud/services/permission_audit_service.py

```diff
@@ -0,0 +1,172 @@
+import datetime
+import logging
+from http import HTTPStatus
+from typing import List
+
+from api.app.constants import (ERROR_CODE_UNKNOWN_STATE,
+                               PrivilegeChangeTypeEnum,
+                               PrivilegeDetailsPermissionTypeEnum,
+                               PrivilegeDetailsScopeTypeEnum)
+from api.app.crud import crud_utils
+from api.app.integration.forest_client_integration import \
+    ForestClientIntegrationService
+from api.app.models.model import FamUser, FamUserRoleXref
+from api.app.repositories.permission_audit_repository import \
+    PermissionAuditRepository
+from api.app.schemas.fam_user_role_assignment_create_response import \
+    FamUserRoleAssignmentCreateRes
+from api.app.schemas.permission_audit_history import \
+    PermissionAduitHistoryCreateSchema
+from api.app.schemas.privilege_change_performer import \
+    PrivilegeChangePerformerSchema
+from api.app.schemas.privilege_details import (PrivilegeDetailsRoleSchema,
+                                               PrivilegeDetailsSchema,
+                                               PrivilegeDetailsScopeSchema)
+from api.app.schemas.requester import RequesterSchema
+from fastapi import HTTPException
+from sqlalchemy.orm import Session
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PermissionAuditService:
+
+    def __init__(self, db: Session):
+        self.repo = PermissionAuditRepository(db)
```

Comment:
This is `server/backend`. Everything was under `crud` folder.
In the past, we always liked to get rid of the `crud` but had no time and would be too big of refactoring at once.
For this new feature, I started using same layers structure (server/repository) from `server/admin_management` to start moving toward consistency with `server/admin_management`.